### PR TITLE
Add implicit return and if-else to overview

### PR DESCRIPTION
**Pull request by kyldvs**
Implicit return is a pretty important concept that is not explicitly mentioned anywhere, adding to the overview

Also split out the if-else section and removed the dedicated page. If-else is such a common language construct and it's not significantly different in Reason so I don't think it needs its own page. A mention in the overview should be fine

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -74,9 +74,34 @@ Comparison                      | `>`, `<`, `>=`, `<=`
 Boolean operations              | `!`, `&&`, <code>&#124;&#124;</code>
 Reference equality              | `===`, `!==`
 Structural equality             | `==`, `!=`
+
+## Implicit Return
+
+There is no `return` keyword in Reason. The last expression in a block or
```
Comment (jordwalke) [inline docs/overview.md:80]: Did we introduce "expression blocks" at this point in the docs yet? If not, I might include it before this.

File: docs/overview.md [hunk 2]
```diff
@@ -74,9 +74,34 @@ Comparison                      | `>`, `<`, `>=`, `<=`
...
+function definition is the return value for that block.
```
Comment (jordwalke) [inline docs/overview.md:81]: I don't think that "return values" make sense for blocks, only for functions. Blocks are expressions that evaluate to a value like every other expression, so it's probably more accurate to say that "The last item in an expression block is what that expression block evaluates to". 

Then somehow explain that functions just have a single expression on the right hand side of their `=>`, and if it is a block so be it - and that expression will evaluate like every other expression, with no ability to early return.

File: docs/overview.md [hunk 3]
```diff
@@ -74,9 +74,34 @@ Comparison                      | `>`, `<`, `>=`, `<=`
...
+returned.
+
+## If-Else Expressions
+
+Feature                         | Example
+--------------------------------|----------
 If-Else expressions             | `if (condition) { a; } else { b; }`
 Ternary expressions             | `condition ? a : b;`
 
+- Note: These are expressions and can be assigned to a variable: `let x = if (condtion) { a; } else { b; };`
```
Comment (jordwalke) [inline docs/overview.md:103]: like-stamp

File: docs/overview.md (hunk 2 above)
Comment (kyldvs) [reply docs/overview.md]: Changed the example to deal with functions so this is less confusing.

File: docs/overview.md (hunk 1 above)
Comment (kyldvs) [reply docs/overview.md]: Removed expression blocks from the example.

**Pull request merged**